climada/engine/test/test_impact_calc.py

```diff
@@ -135,6 +135,32 @@ def test_error_handling_mismatch_haz_type(self):
         except Exception as e:
             self.assertEqual(str(e), "Impact calculation not possible. No impact "
                              "functions found for hazard type TC in impf_set.")
+    def test_error_handling_mismatch_impf_ids(self):
+        """Test error handling in case impf ids in exposures
+        does not appear in impf_set"""
+        haz = Hazard('TC')
+        exp = Exposures()
+        exp.gdf.loc[0,'impf_TC'] = 1
+        exp.gdf.loc[1,'impf_TC'] = 2
+        impf_exp = ImpactFunc()
+        impf_exp.id = 1
+        impf_exp.intensity = np.array([0, 20])
+        impf_exp.paa = np.array([0, 1])
+        impf_exp.mdd = np.array([0, 0.5])
+        impf_exp.haz_type = 'TC'
+        impf_noexp = deepcopy(impf_exp)
+        impf_noexp.id = 3
+        impfset = ImpactFuncSet([impf_exp, impf_noexp])
+
+        try:
+            ImpactCalc(exp, impfset, haz).impact()
+        except Exception as e:
+            self.assertEqual(str(e),  "At least one impact function associated to"
+                             " the exposures has no match in the impact function set.\n"
+                "The impact functions in the impact function set for hazard type \"TC\" "
+                "have ids [1, 3]. The column \"impf_TC\" in the exposures contains the ids"
+                " [1, 2].\nPlease make sure that all exposure points are associated with an "
+                "impact function that is included in the impact function set.")
```

Comment:
Sorry, I did not see that first. For the unit test, please use `self.assertRaises()` : https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises

Reply:
Yes that makes sense to use this instead. I modified the testing function.